Failure with Intel ICPC 19.1.3 on MacOS

I am getting the following error following the examples on the Readme:
```
path/variant_example/3rdparty/variant/include/mpark/variant.hpp(1613): error: parameter pack "Ts" was referenced but not expanded
          typename Front = lib::type_pack_element_t<0, Ts...>,
                           ^

compilation aborted for path/variant_example/hello_world.cpp (code 2)
make[2]: *** [CMakeFiles/hello-world.dir/hello_world.cpp.o] Error 2
make[1]: *** [CMakeFiles/hello-world.dir/all] Error 2
```
The intel C++ compiler ties to the underlying apple-clang compiler and I can see that icpc is setting "___clang___".
```
cheddar:variant_example: $ icpc -dM -E -std=c++11 -x c++ -I 3rdparty/variant/include ./hello_world.cpp | grep -yR __clang__
(standard input):#define __clang__ 1
(standard input):#define __clang__ 1
```

I'm not sure if this is an Intel error or not since by default Intel ties to the underlying apple-clang C/C++ compiler.

Thanks for looking, 
Tony

--- a/include/mpark/config.hpp
+++ b/include/mpark/config.hpp
@@ -50,7 +50,7 @@
 #define MPARK_BUILTIN_UNREACHABLE
 #endif
 
-#if __has_builtin(__type_pack_element)
+#if __has_builtin(__type_pack_element) && !(defined(__ICC))
 #define MPARK_TYPE_PACK_ELEMENT
 #endif
 

